corehq/apps/app_manager/tests/test_schedule.py

```diff
@@ -0,0 +1,381 @@
+                ScheduleVisit(due=11)
+            ]
+        )
+        other_module.forms.append(copy.copy(scheduled_form))
+
+        other_module.schedule_phases = [
+            SchedulePhase(
+                anchor='case_property',
+                forms=[SchedulePhaseForm(form_id=scheduled_form.unique_id)]
+            )
+        ]
+
+        expected_fixture = """
+             <partial>
+             <fixture id="schedule:m2:p1:f0">
+                 <schedule expires="" allow_unscheduled="False">
+                     <visit id="1" due="9" repeats="False"/>
+                     <visit id="2" due="11" repeats="False"/>
+                 </schedule>
+             </fixture>
+             </partial>
+        """
+
+        suite = self.app.create_suite()
+
+        self.assertXmlPartialEqual(expected_fixture, suite, './fixture[@id="schedule:m2:p1:f0"]')
+        self.assertXmlHasXpath(suite, './fixture[@id="schedule:m1:p1:f0"]')
+
+    def test_form_filtering(self):
+        self._apply_schedule_phases()
+        suite = self.app.create_suite()
+        form_ids = (self.form_1.schedule_form_id, self.form_2.schedule_form_id)
+        case_load_actions = ["case_id_case_clinic", "case_id_load_clinic0"]
+        case = ["instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/{}]"
+                .format(action) for action in case_load_actions]
+        for form_num, form_id in enumerate(form_ids):
+            anchor = "{case}/edd".format(case=case[form_num])
+            current_schedule_phase = "{case}/current_schedule_phase".format(case=case[form_num])
+            visit = "instance('schedule:m1:p1:f{form_num}')/schedule/visit".format(form_num=form_num)
+            schedule = "instance('schedule:m1:p1:f{form_num}')/schedule".format(form_num=form_num)
+
+            filter_condition = (
+                "(({current_schedule_phase} = '' or {current_schedule_phase} = 1) "  # form phase == current phase
+                "and {anchor} != '' and "                # anchor not empty
+                "(today() &gt;= (date({anchor}) + int({schedule}/@starts)) and "
+                "({schedule}/@expires = '' or today() &lt;= (date({anchor}) + int({schedule}/@expires))))) and "
+                "({schedule}/@allow_unscheduled = 'True' or "
+                "count({visit}[{case}/last_visit_number_{form_id} = '' or "
+                "@id &gt; {case}/last_visit_number_{form_id}]["
+                "if(@repeats = 'True', "
+                "today() &gt;= (date({case}/last_visit_date_{form_id}) + int(@increment) + int(@starts)) and "
+                "(@expires = '' or "
+                "today() &lt;= (date({case}/last_visit_date_{form_id}) + int(@increment) + int(@expires))), "
+                "today() &gt;= (date({anchor}) + int(@due) + int(@starts)) and "
+                "(@expires = '' or today() &lt;= (date({anchor}) + int(@due) + int(@expires))))"
+                "]) &gt; 0)"     # End count
```

Comment:
hah! I'm going to break it out into separate variables so it reads better, and matches the `SchedulerXPath` properties that these get derived from.
eg: 
`within_window = "today() >= (date({anchor}) + int({schedule}/@starts)"..."` 
etc. 
I think the one in the test fixture file is actually more unwieldy since you can't add linebreaks, and it is just a huge blob of xpath that I am testing.